src/patients/GeneralInformation.tsx

```diff
@@ -160,12 +156,12 @@ const GeneralInformation = (props: Props) => {
                     ? new Date(patient.dateOfBirth)
                     : undefined
                 }
-                isInvalid={error?.dateOfBirth}
                 maxDate={new Date(Date.now().valueOf())}
-                feedback={t(error?.dateOfBirth)}
-                onChange={(date: Date) => {
-                  onDateOfBirthChange(date)
-                }}
+                onChange={(date: Date) => onFieldChange('dateOfBirth', date.toISOString())}
+                isInvalid={!!error?.dateOfBirth}
+                feedback={
+                  error ? (error.dateOfBirth ? t(error.dateOfBirth) : undefined) : undefined
```

Comment:
we do this sort of login on a lot of spots... lets make a helper hook. 


```
// useTranslation.ts
import { useTranslation } from 'react-i18next'

export default function useTranslator() {
  const { t } = useTranslation()

  function translate(key: undefined): undefined
  function translate(key: string): string
  function translate(key: any): any {
    return key !== undefined ? t(key) : undefined
  }

  return {
    t: translate,
  }
}
```

it would be used like: 


```
const { t } = useTranslator()
```

and we can replace 

`feedback={error ? (error.dateOfBirth ? t(error.dateOfBirth) : undefined) : undefined`}
with
`feedback={t(error?.dateOfBirth)}`

Reply:
Opened a ticket: https://github.com/HospitalRun/hospitalrun-frontend/issues/2135